### PR TITLE
Feature:  Update logic on adding and removing a collator

### PR DESCRIPTION
**Pull request by thadouk**
Updates the logic and tests of the validators manager pallet, to align with the smart contract interface.
Updates the way the confirmations are generated, reactors tests and benchmarks around that area.

Related JIRA tasks:
- SYS-3387
- SYS-3333
- SYS-3360

### CODE REVIEW

**Review by ivan-cholakov — APPROVED** [review]

File: pallets/ethereum-transactions/src/ethereum_transaction.rs [hunk 1]
```diff
@@ -33,9 +41,8 @@ impl EthTransactionType {
     pub fn to_abi(&self) -> Result<EthTransactionDescription, ethabi::Error> {
         match self {
             EthTransactionType::PublishRoot(d) => Ok(d.to_abi()),
-            EthTransactionType::DeregisterValidator(d) => Ok(d.to_abi()),
-            EthTransactionType::SlashValidator(d) => Ok(d.to_abi()),
-            EthTransactionType::ActivateValidator(d) => Ok(d.to_abi()),
+            EthTransactionType::ActivateCollator(d) => Ok(d.to_abi()),
+            EthTransactionType::DeregisterCollator(d) => Ok(d.to_abi()),
```
Comment (nahuseyoum) [inline pallets/ethereum-transactions/src/ethereum_transaction.rs:45]: Is "re-enabling slashing" captured anywhere in JIRA? We seem to be removing that functionality here.

File: pallets/validators-manager/src/confirmations.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
+use frame_support::log;
+use pallet_ethereum_transactions::ethereum_transaction::{
+    ActivateCollatorData, DeregisterCollatorData,
+};
+use sp_io::hashing::keccak_256;
+
+const PACKED_KEYS_SIZE: usize = 96;
+
+#[cfg(test)]
```
Comment (nahuseyoum) [inline pallets/validators-manager/src/confirmations.rs:9]: Do we want this here?

File: pallets/validators-manager/src/tests_voting_deregistration.rs [hunk 3]
```diff
@@ -93,24 +106,36 @@ fn setup_voting_session(action_id: &ActionId<AccountId>) {
     assert_eq!(ValidatorManager::get_vote(action_id).nays.is_empty(), true);
 }
 
+// TODO use the private keys of the authorities to sign the _eth_compatible_data
+fn create_valid_signed_respose(
```
Comment (nahuseyoum) [inline pallets/validators-manager/src/tests_voting_deregistration.rs:110]:
```suggestion
fn create_valid_signed_response(
```

**Review by nahuseyoum — APPROVED** [review]
LGTM but we should make a decision on the Slashing logic. I think we are implicitly removing it and I am not sure if that is a good idea.

File: pallets/ethereum-transactions/src/tests/tests_eth_transaction_type.rs [hunk 4]
```diff
@@ -2,16 +2,21 @@
 
 use crate::ethereum_transaction::*;
 use ethabi::{Function, Param, ParamType, Token};
+use sp_core::H512;
 
 pub const ROOT_HASH: [u8; 32] = [3; 32];
 pub const T2_PUBLIC_KEY: [u8; 32] = [4; 32];
+pub const T1_PUBLIC_KEY: [u8; 64] = [5u8; 64];
```
Comment (fluorostani) [inline pallets/ethereum-transactions/src/tests/tests_eth_transaction_type.rs:9]: we should be consistent over whether we specify the type or not here

**Review by fluorostani — APPROVED** [review]

File: pallets/ethereum-transactions/src/ethereum_transaction.rs (hunk 1 above)
Comment (thadouk) [reply pallets/ethereum-transactions/src/ethereum_transaction.rs:45]: yes we have a ticket for it.

File: pallets/validators-manager/src/confirmations.rs (hunk 2 above)
Comment (nahuseyoum) [reply pallets/validators-manager/src/confirmations.rs:9]: Move it to lib.rs with the rest of the tests

Comment (thadouk) [reply pallets/validators-manager/src/confirmations.rs:9]: Why? The context is much better this way. This file has this set of tests for these functions.

File: pallets/ethereum-transactions/src/tests/tests_eth_transaction_type.rs (hunk 4 above)
Comment (thadouk) [reply pallets/ethereum-transactions/src/tests/tests_eth_transaction_type.rs:9]: I think I get it now, you mean `[5u8; 64]` vs `[5; 64]`
I prefer the first way, it's more explicit.

File: pallets/validators-manager/src/confirmations.rs (hunk 2 above)
Comment (nahuseyoum) [reply pallets/validators-manager/src/confirmations.rs:9]: Having these paths scattered all over the place will make it harder to see where they are. What is the problem of having them in 1 place, grouped together? What is the rational to change this existing practise?

Comment (thadouk) [reply pallets/validators-manager/src/confirmations.rs:9]: The rationale behind this proposal is to improve modularity within our codebase. Currently, the lib.rs file is responsible for organizing modules and tests. Instead, a simpler approach would be for each module to specify the path to its test files as a self-contained unit. It is a more modular approach I think.

**Pull request merged**